src/common/AndroidUtils.ts

```diff
@@ -362,106 +339,88 @@ export class AndroidSDKUtils {
         const timeout = androidConfig.deviceBootReadinessWaitTime;
-        const numberOfRetries = androidConfig.deviceBootStatusPollRetries;
-        return new Promise<boolean>((resolve, reject) => {
-            const timeoutFunc = (commandStr: string, noOfRetries: number) => {
-                try {
-                    const stdout = CommonUtils.executeCommandSync(commandStr);
-                    if (stdout && stdout.trim() === '1') {
-                        resolve(true);
-                    } else {
-                        if (noOfRetries === 0) {
-                            reject(
-                                new Error(
-                                    `Timeout waiting for emulator-${portNumber} to boot.`
-                                )
-                            );
-                        } else {
-                            setTimeout(
-                                timeoutFunc,
-                                timeout,
-                                commandStr,
-                                noOfRetries - 1
-                            );
-                        }
-                    }
-                } catch (error) {
-                    reject(
-                        new Error(
-                            `Unable to communicate with emulator via ADB.`
-                        )
-                    );
+        let numberOfRetries = androidConfig.deviceBootStatusPollRetries;
+
+        while (numberOfRetries > 0) {
+            try {
+                const stdout = CommonUtils.executeCommandSync(command);
+                if (stdout && stdout.trim() === '1') {
+                    return Promise.resolve();
+                } else {
+                    await CommonUtils.delay(timeout);
+                    numberOfRetries--;
                 }
-            };
-            setTimeout(timeoutFunc, 1000, command, numberOfRetries);
-        });
+            } catch (error) {
+                return Promise.reject(
+                    new Error(
+                        `Unable to communicate with emulator via ADB: ${error}`
+                    )
+                );
+            }
+        }
+
+        return Promise.reject(
+            new Error(`Timeout waiting for emulator-${portNumber} to boot.`)
+        );
     }
 
-    public static launchURLIntent(
+    public static async launchURLIntent(
         url: string,
         emulatorPort: number
-    ): Promise<boolean> {
+    ): Promise<void> {
         const openUrlCommand = `${AndroidSDKUtils.getAdbShellCommand()} -s emulator-${emulatorPort} shell am start -a android.intent.action.VIEW -d ${url}`;
-        return new Promise((resolve, reject) => {
-            try {
-                CommonUtils.executeCommandSync(openUrlCommand);
-                resolve(true);
-            } catch (error) {
-                reject(error);
-            }
-        });
+        return CommonUtils.executeCommandAsync(openUrlCommand)
```

Comment:
This should just be `return CommonUtils.executeCommandAsync(openUrlCommand);`, right? The `then` and `catch` blocks below are just pass-through.

Reply:
Oh never mind that's going to get into casting errors with the Promise signatures.

Reply:
Not exactly. This functions returns `Promise<void>` but `executeCommandAsync` doesn't. So I need to keep the `then` but the `catch` can go.